### PR TITLE
ANN: Make public fix

### PR DESCRIPTION
**Pull request by shevtsiv**

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt [hunk 1]
```diff
@@ -163,16 +164,16 @@ class RsErrorAnnotator : RsAnnotatorBase(), HighlightRangeExtension {
     }
 
     private fun checkReferenceIsPublic(ref: RsReferenceElement, o: PsiElement, holder: AnnotationHolder) {
-        val element = ref.reference.resolve() as? RsVisible ?: return
+        val element = ref.reference.resolve() as? RsVisibilityOwner ?: return
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt]: Looks like it's wrong assumption (e.g. RsFile is RsVisible but not RsVisibilityOwner)

File: src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt [hunk 2]
```diff
@@ -163,16 +164,16 @@ class RsErrorAnnotator : RsAnnotatorBase(), HighlightRangeExtension {
...
         val oMod = o.contextStrict<RsMod>() ?: return
         if (element.isVisibleFrom(oMod)) return
-
+        val withinOneCrate = element.crateRoot == (o as RsElement).crateRoot
         val error = when {
             element is RsNamedFieldDecl -> {
                 val structName = element.ancestorStrict<RsStructItem>()?.crateRelativePath?.removePrefix("::") ?: ""
-                RsDiagnostic.StructFieldAccessError(ref, ref.referenceName, structName)
+                RsDiagnostic.StructFieldAccessError(ref, ref.referenceName, structName, MakePublicFix.createIfCompatible(element, withinOneCrate))
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt]: Please, create the fix inside `RsDiagnostic.StructFieldAccessError` inside `prepare` method

File: src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt [hunk 3]
```diff
@@ -166,13 +167,15 @@ class RsErrorAnnotator : RsAnnotatorBase(), HighlightRangeExtension {
         val element = ref.reference.resolve() as? RsVisible ?: return
         val oMod = o.contextStrict<RsMod>() ?: return
         if (element.isVisibleFrom(oMod)) return
-
+        val withinOneCrate = element.crateRoot == (o as RsElement).crateRoot
```
Comment (ortem) [inline src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt]: Let's change the type of `o` to `RsElement` in the function signature and use `o.crateRoot`

Comment (shevtsiv) [reply src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt]: Thanks, fixed.

Comment (ortem) [reply src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt]: Great! Don't forget to move fixes creations to `prepare` methods (see the unresolved comment).

File: src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt (hunk 2 above)
Comment (vlad20012) [reply src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt]: (I fount that a lot of things already made wrong, and so it's hard and not reasonable to move quickfixes creation into `prepare`...)

**Comment by ortem** [comment]
Hi @shevtsiv, 

Great work, thank you! I'm going to merge this PR if you have finished it. As @vlad20012 mentioned, you actually can leave the fixes creation as is because of existing design problems.

**Comment by shevtsiv** [comment]
@ortem , I finished working on it, so it can be freely merged, thank you.

**Comment by ortem** [comment]
Welcome to IntelliJ Rust contributors!

bors r+

**Pull request merged**